From abf906d35843be10a9321a43062c8f9e411ccc1b Mon Sep 17 00:00:00 2001
From: Frantisek Sumsal <frantisek@sumsal.cz>
Date: Wed, 9 Feb 2022 22:35:03 +0100
Subject: [PATCH] test: lvm 2.03.15 dropped the static autoactivation

so install the respective generator only if we're running with older
lvm versions.

See: https://sourceware.org/git/?p=lvm2.git;a=commit;h=ee8fb0310c53ed003a43b324c99cdfd891dd1a7c
(cherry picked from commit d10d562bd4b9f93130fb2b23f2b0d0d4126ea7d4)

Related: #2087677
---
 test/test-functions | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/test/test-functions b/test/test-functions
index 4827b6bedf..dcc893733c 100644
--- a/test/test-functions
+++ b/test/test-functions
@@ -973,16 +973,19 @@ install_lvm() {
     image_install lvm
     image_install "${ROOTLIBDIR:?}"/system/lvm2-lvmpolld.{service,socket}
     image_install "${ROOTLIBDIR:?}"/system/{blk-availability,lvm2-monitor}.service
-    image_install "${ROOTLIBDIR:?}"/system-generators/lvm2-activation-generator
     image_install -o "/lib/tmpfiles.d/lvm2.conf"
     if get_bool "$LOOKS_LIKE_DEBIAN"; then
         inst_rules 56-lvm.rules 69-lvm-metad.rules
     else
         # Support the new udev autoactivation introduced in lvm 2.03.14
         # https://sourceware.org/git/?p=lvm2.git;a=commit;h=67722b312390cdab29c076c912e14bd739c5c0f6
+        # Static autoactivation (via lvm2-activation-generator) was dropped
+        # in lvm 2.03.15
+        # https://sourceware.org/git/?p=lvm2.git;a=commit;h=ee8fb0310c53ed003a43b324c99cdfd891dd1a7c
         if [[ -f /lib/udev/rules.d/69-dm-lvm.rules ]]; then
             inst_rules 11-dm-lvm.rules 69-dm-lvm.rules
         else
+            image_install "${ROOTLIBDIR:?}"/system-generators/lvm2-activation-generator
             image_install "${ROOTLIBDIR:?}"/system/lvm2-pvscan@.service
             inst_rules 11-dm-lvm.rules 69-dm-lvm-metad.rules
         fi
